share/eds/byzantine/bad_encoding_test.go

```diff
@@ -3,19 +3,45 @@ package byzantine
 import (
 	"context"
 	"testing"
+	"time"
 
 	mdutils "github.com/ipfs/go-merkledag/test"
+	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 	core "github.com/tendermint/tendermint/types"
 
 	"github.com/celestiaorg/celestia-app/pkg/da"
 	"github.com/celestiaorg/rsmt2d"
 
 	"github.com/celestiaorg/celestia-node/header"
-	"github.com/celestiaorg/celestia-node/share"
+	"github.com/celestiaorg/celestia-node/share/eds/edstest"
 	"github.com/celestiaorg/celestia-node/share/ipld"
+	"github.com/celestiaorg/celestia-node/share/sharetest"
 )
 
+func TestBadEncodingFraudProof(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*15)
+	defer t.Cleanup(cancel)
+	bServ := mdutils.Bserv()
+
+	square := edstest.RandByzantineEDS(t, 16)
+	dah := da.NewDataAvailabilityHeader(square)
+	err := ipld.ImportEDS(ctx, square, bServ)
+	require.NoError(t, err)
+
+	var errRsmt2d *rsmt2d.ErrByzantineData
+	err = square.Repair(dah.RowRoots, dah.ColumnRoots)
+	require.ErrorAs(t, err, &errRsmt2d)
+
+	errByz := NewErrByzantine(ctx, bServ, &dah, errRsmt2d)
+
+	befp := CreateBadEncodingProof([]byte("hash"), 0, errByz)
+	err = befp.Validate(&header.ExtendedHeader{
+		DAH: &dah,
+	})
+	assert.NoError(t, err)
+}
```

Comment:
It's moved from Retriever tests to avoid cycle. It's also the right location for it.